lib/read.js

```diff
@@ -128,7 +125,21 @@ function read (req, res, next, parse, debug, options) {
     }
 
     next()
-  })
+  }
+
+  debug('read body')
+  if (typeof global.Promise === 'function') {
+    getBody(stream, opts)
+      .then(readSuccessHandler)
+      .catch(readErrorHandler)
```

Comment:
```suggestion
      .then(readSuccessHandler, readErrorHandler)
```

Should it be written this way? From my understanding on promises, the way it is written means `readErrorHandler` will run for errors in `readSuccessHandler` too, but it was only written to handle the errors from the `getBody` itself.

Reply:
You're absolutely right. I've been working with the `async/await` flavour of promises for a while now and I'm beginning to forget the intricacies of raw promises :sweat_smile: Thanks!